internal/kafka/kafka.go

```diff
@@ -123,8 +124,22 @@ func NewKafkaBroker(ctx context.Context) (Broker, error) {
 	}, nil
 }
 
+// kafka library has some noisy debug messages
+var ignoredMsg *regexp.Regexp
+
+func init() {
+	var err error
+	ignoredMsg, err = regexp.Compile("^no messages received from kafka within the allocated time.*")
+	if err != nil {
+		panic(err)
+	}
+}
+
 func newContextLogger(ctx context.Context) func(msg string, a ...interface{}) {
```

Comment:
I do not see any usage of `newContextLogger` function, to ignore those messages the function needs to be used throughout the code.

Reply:
It is here:

```
// NewReader creates a reader. Use Close() function to close the reader.
func (b *kafkaBroker) NewReader(ctx context.Context, topic string) *kafka.Reader {
	return kafka.NewReader(kafka.ReaderConfig{
		Brokers:     config.Kafka.Brokers,
		Dialer:      b.dialer,
		Topic:       topic,
		StartOffset: kafka.LastOffset,
		Logger:      kafka.LoggerFunc(newContextLogger(ctx)),
		ErrorLogger: kafka.LoggerFunc(newContextErrLogger(ctx)),
	})
}

// NewWriter creates synchronous writer created from the pool. It does not have associated any topic with it,
// therefore topic must be set on the message-level. Make sure to close it with Close() function.
func (b *kafkaBroker) NewWriter(ctx context.Context) *kafka.Writer {
	return &kafka.Writer{
		Addr:        kafka.TCP(config.Kafka.Brokers...),
		Transport:   b.transport,
		Logger:      kafka.LoggerFunc(newContextLogger(ctx)),
		ErrorLogger: kafka.LoggerFunc(newContextErrLogger(ctx)),
	}
}
```

Reply:
oh, missed it, it showed only one result in the search.